spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/core/CredentialsProviderAutoConfigurationTests.java

```diff
@@ -89,6 +96,42 @@ void credentialsProvider_profileNameAndPathConfigured_configuresProfileCredentia
 				});
 	}
 
+	@Test
+	void credentialsProvider_stsPropertiesConfigured_configuresStsWebIdentityTokenFileCredentialsProvider()
+			throws IOException {
+		File tempFile = tokenTempDir.resolve("token-file.txt").toFile();
+		tempFile.createNewFile();
+
+		this.contextRunner
+				.withPropertyValues("spring.cloud.aws.region.static:af-south-1",
+						"spring.cloud.aws.credentials.sts.role-arn:develop",
+						"spring.cloud.aws.credentials.sts.web-identity-token-file:" + tempFile.getAbsolutePath())
+				.run((context) -> {
+					AwsCredentialsProvider awsCredentialsProvider = context.getBean("credentialsProvider",
+							AwsCredentialsProvider.class);
+					assertThat(awsCredentialsProvider).isNotNull()
+							.isInstanceOf(StsWebIdentityTokenFileCredentialsProvider.class);
+				});
+	}
+
+	@Test
+	void credentialsProvider_stsSystemPropertiesDefault_configuresStsWebIdentityTokenFileCredentialsProvider()
```

Comment:
This test is failing, for a good reason - `spring.cloud.aws.credentials.sts.enabled` has been removed.

Look another comment